### DISCUSSION TITLE
How to print color dicom in web app by weasis

### DISCUSSION THREAD

**Discussion by Georgeqhh** [nroduit/Weasis#382] [Q&A]
In my web app,I use weasis to print dicom,when it is not color print, it is normal,when it is color print,will get the following error,but it is normal when I use weasis app that built  from source code
A-ASSOCIATE-RJ[result: 1 - rejected-permanent, source: 2 - service-provider (ACSE related function), reason: 1 - no-reason-given]
	at org.dcm4che3.net.PDUDecoder.nextPDU(PDUDecoder.java:182)
	at org.dcm4che3.net.Association$2.run(Association.java:571)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
How could I know the failure reason and the solution

**Comment by nroduit** [comment]
DICOM Print is an external contribution and it is difficult for me to maintain this feature because in our hospital we don't use it anymore. Moreover, there are often proprietary aspects with DICOM Print.

Unfortunately, I can't investigate this problem because you need the printer to reproduce it and maybe the problem is also in the limitations or configuration of the printer.

**Reply by Georgeqhh** [reply]
These logs can help you guess the more possible reason?
Printer log:
MIM_CORE 10712 03/21/23 14:28:04 The association request is rejected. - No presentation context items accepted 

Weasis Log:
20.03.2023 10:27:19.727 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Connection: Initiate connection from 0.0.0.0/0.0.0.0:0 to 192.168.40.48:5040
20.03.2023 10:27:19.727 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Connection: Established connection Socket[addr=/192.168.40.48,port=5040,localport=63847]
20.03.2023 10:27:19.734 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Association: WEASIS_AE->DV5700(1) << A-ASSOCIATE-RQ
20.03.2023 10:27:19.990 *INFO* [pool-6-thread-1] org.dcm4che3.net.Association: WEASIS_AE->DV5700(1) >> A-ASSOCIATE-AC
20.03.2023 10:27:19.999 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 1:N-CREATE-RQ[pcid=1
  cuid=1.2.840.10008.5.1.1.1 - Basic Film Session SOP Class
  iuid=2.25.273264796544878771126571609696422720646 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:20.058 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 1:N-CREATE-RSP[pcid=1, status=0H
  cuid=1.2.840.10008.5.1.1.1 - Basic Film Session SOP Class
  iuid=2.25.273264796544878771126571609696422720646 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:20.109 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 2:N-CREATE-RQ[pcid=1
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:20.188 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 2:N-CREATE-RSP[pcid=1, status=0H
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:20.234 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 3:N-SET-RQ[pcid=1
  cuid=1.2.840.10008.5.1.1.4 - Basic Grayscale Image Box SOP Class
  iuid=1.2.840.113564.11.192.168.40.48.2023.3.20.10.29.46.44 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.033 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 3:N-SET-RSP[pcid=1, status=0H
  cuid=1.2.840.10008.5.1.1.4 - Basic Grayscale Image Box SOP Class
  iuid=1.2.840.113564.11.192.168.40.48.2023.3.20.10.29.46.44 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.033 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 4:N-ACTION-RQ[pcid=1, actionID=1
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.105 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 4:N-ACTION-RSP[pcid=1, actionID=1, status=0H
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.105 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 5:N-DELETE-RQ[pcid=1
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.108 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 5:N-DELETE-RSP[pcid=1, status=0H
  cuid=1.2.840.10008.5.1.1.2 - Basic Film Box SOP Class
  iuid=2.25.311976451276306843289453615290236834482 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.109 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) << 6:N-DELETE-RQ[pcid=1
  cuid=1.2.840.10008.5.1.1.1 - Basic Film Session SOP Class
  iuid=2.25.273264796544878771126571609696422720646 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.115 *INFO* [pool-6-thread-1] org.dcm4che3.net.Dimse: WEASIS_AE->DV5700(1) >> 6:N-DELETE-RSP[pcid=1, status=0H
  cuid=1.2.840.10008.5.1.1.1 - Basic Film Session SOP Class
  iuid=2.25.273264796544878771126571609696422720646 - ?
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian]
20.03.2023 10:27:22.115 *INFO* [AWT-EventQueue-0] org.dcm4che3.net.Association: WEASIS_AE->DV5700(1) << A-RELEASE-RQ
20.03.2023 10:27:22.115 *INFO* [pool-6-thread-1] org.dcm4che3.net.Association: WEASIS_AE->DV5700(1) >> A-RELEASE-RP
20.03.2023 10:27:22.115 *INFO* [pool-6-thread-1] org.dcm4che3.net.Association: WEASIS_AE->DV5700(1): close Socket[addr=/192.168.40.48,port=5040,localport=63847]

Web app log:
2023-03-20 15:40:35.500 [http-nio-8087-exec-2] INFO  o.d.net.Connection - [connect,1018] - Initiate connection from 0.0.0.0/0.0.0.0:0 to 192.168.40.48:5040
2023-03-20 15:40:35.501 [http-nio-8087-exec-2] INFO  o.d.net.Connection - [connect,1053] - Established connection Socket[addr=/192.168.40.48,port=5040,localport=63068]
2023-03-20 15:40:35.539 [http-nio-8087-exec-2] DEBUG o.d.net.Association - [enterState,534] - /192.168.50.62:63068->/192.168.40.48:5040(1): enter state: Sta4 - Awaiting transport connection opening to complete
2023-03-20 15:40:35.558 [http-nio-8087-exec-2] INFO  o.d.net.Association - [write,507] - WEASIS_AE->DV5700(1) << A-ASSOCIATE-RQ
2023-03-20 15:40:35.558 [http-nio-8087-exec-2] DEBUG o.d.net.Association - [write,508] - A-ASSOCIATE-RQ[
  calledAET: DV5700
  callingAET: WEASIS_AE
  applicationContext: 1.2.840.10008.3.1.1.1 - DICOM Application Context Name
  implClassUID: 1.2.40.0.13.1.3
  implVersionName: dcm4che-5.25.2
  maxPDULength: 16378
  maxOpsInvoked/maxOpsPerformed: 1/1
  PresentationContext[id: 1
    as: 1.2.840.10008.5.1.1.18 - Basic Color Print Management Meta SOP Class
    ts: 1.2.840.10008.1.2 - Implicit VR Little Endian
  ]
]
2023-03-20 15:40:35.564 [http-nio-8087-exec-2] DEBUG o.d.net.Association - [enterState,534] - WEASIS_AE->DV5700(1): enter state: Sta5 - Awaiting A-ASSOCIATE-AC or A-ASSOCIATE-RJ PDU
2023-03-20 15:40:35.619 [pool-7-thread-1] INFO  o.d.net.Association - [onAAssociateRJ,654] - WEASIS_AE->DV5700(1) >> A-ASSOCIATE-RJ[result: 1 - rejected-permanent, source: 2 - service-provider (ACSE related function), reason: 1 - no-reason-given]
2023-03-20 15:40:35.620 [pool-7-thread-1] INFO  o.d.net.Association - [doCloseSocket,356] - WEASIS_AE->DV5700(1): close Socket[addr=/192.168.40.48,port=5040,localport=63068]
2023-03-20 15:40:35.621 [pool-7-thread-1] DEBUG o.d.net.Association - [enterState,534] - WEASIS_AE->DV5700(1): enter state: Sta1 - Idle
A-ASSOCIATE-RJ[result: 1 - rejected-permanent, source: 2 - service-provider (ACSE related function), reason: 1 - no-reason-given]
	at org.dcm4che3.net.PDUDecoder.nextPDU(PDUDecoder.java:182)
	at org.dcm4che3.net.Association$2.run(Association.java:571)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by Georgeqhh** [comment]
I try color print in weasis,it is not ok also,so maybe not select the color print in weases last time

**Comment by nroduit** [comment]
From the error, it seems to fail to negotiate the SopClassUID for the color. However, the error is not very explicit.

Does the printer support the SopClassUID BasicColorPrintManagementMeta = "1.2.840.10008.5.1.18"? Look in the DICOM Conformance Statement.

**Reply by Georgeqhh** [reply]
It doesn't support color print,so it should not support the SopClassUID BasicColorPrintManagementMeta = "1.2.840.10008.5.1.18",but it doesn't work well in one printer that supports color print,and my fellow dosen't give me the log,next week I will find a color printer to test it.Thank you for your help.
